### ISSUE TITLE
Confusing arg name for sequence_loss

### ISSUE THREAD

**Issue by Conchylicultor** [tensorflow/tensorflow#6937]
Sorry if this is noise, but it's just a minor suggestion to make the API better (before the release of TensorFlow 1.0 where it will be much more difficult to change)

For the function [sequence_loss](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/seq2seq/python/ops/loss.py) on the master branch, the name of the parameter `softmax_loss_function` seems to imply that only softmax based loss are compatible which is not true (for instance `tf.nn.sigmoid_cross_entropy_with_logits` works too). Maybe the name was that just to indicate the default behavior but I think it is misleading.

Why not simply call the parameter `loss_function` ? It's more representative of what the parameter really does.

**Comment by asimshankar** [comment]
`contrib` and its submodules are not covered by API stability guarantees, so we do have an opportunity to change them. This was introduced in PR #4382 

@lukaszkaiser @ebrevdo : Any opinions here?

**Comment by ebrevdo** [comment]
I think all that code belonged in seq2seq_legacy, it doesn't seem to really
be a good fit for the new seq2seq API.

**Comment by gunan** [comment]
Automatically closing due to lack of recent activity.
please feel free to reopen if this is still an issue, or better yet, send a pull request with the proposed fix.
